week2/project/ecommerce/src/pages/SingleProductPage/SingleProductPage.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { useParams } from 'react-router-dom'
+import SingleProduct from '../../components/product/SingleProduct'
+import './singleProductPage.css'
+import ErrorMessage from '../../components/common/ErrorMessage';
+import Spinner from '../../components/common/Spinner/Spinner';
+
+export default function ProductDatail() {
+   const { itemId } = useParams();
+   const [errorObj, setErrorObj] = React.useState({ isError: false, message: '' })
+   const [itemInfo, setItemInfo] = React.useState([]);
+   const [isLoading, setIsLoading] = React.useState(false);
```

Comment:
nice and it's easier also to import useState form react to use it directly instead using React.useState everytime,